src/in_process_server.cpp

```diff
@@ -734,7 +734,7 @@ class wlcs::Client::Impl
         auto xdg_surface = std::make_shared<XdgSurfaceV6>(client, surface);
         auto xdg_toplevel = std::make_shared<XdgToplevelV6>(*xdg_surface);
 
-        run_on_destruction([xdg_surface, xdg_toplevel]() mutable
+        surface.run_on_destruction([xdg_surface, xdg_toplevel]() mutable
```

Comment:
This would really be clearer with `FnOnce`, wouldn't it. We can't *actually* make these `unique_ptr`s, because we can't get the compiler to move-only the lambda?

Reply:
I guess it might be a *little* clearer if you moved the `shared_ptr`s into the lambda capture, but I'm not going to block for that.